backend/engine/plugins/gitsecrets/secrets_processor.py

```diff
@@ -2,56 +2,150 @@
-    re.compile(str(r"(AKIA[A-Z0-9]{16})")),
-    re.compile(str(r"(ASIA[A-Z0-9]{16})")),
-    re.compile(str(r"(ACCA[A-Z0-9]{16})")),
-    re.compile(str(r"(EAACEdEose0cBA[0-9A-Za-z]{16,255})")),
-    re.compile(str(r"([f|F][a|A][c|C][e|E][b|B][o|O][o|O][k|K](.){1,64}['|\"][0-9a-f]{32}['|\"])")),
-    re.compile(str(r"([a|A][p|P][i|I][_]?[k|K][e|E][y|Y](.){1,64}['|\"][0-9a-zA-Z]{32,45}['|\"])")),
-    re.compile(str(r"([s|S][e|E][c|C][r|R][e|E][t|T](.){1,64}['|\"][0-9a-zA-Z]{32,45}['|\"])")),
-    re.compile(str(r"([0-9]+-[0-9A-Za-z_]{32}\.apps\.googleusercontent\.com)")),
-    re.compile(str(r"(ya29\.[0-9A-Za-z\-_]{55})")),
-    re.compile(
-        str(
-            (
-                r"((ftp|ftps|http|https|mongodb|mongodb\+srv|postgres)"
-                r"://[^/\s:@]{3,64}:[^/\s:@]{3,64}@[a-zA-Z0-9\.-]+(:[0-9]{1,5}){0,1})"
+    SecretRegex(
+        finding_type="Amazon MWS",
+        regex=re.compile(str(r"(amzn\.mws\.[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")),
+    ),
+    SecretRegex(
+        finding_type="google",
+        regex=re.compile(str(r"(AIza[0-9A-Za-z_-]{35})")),
+    ),
+    SecretRegex(
+        finding_type="slack",
+        regex=re.compile(str(r"(https://hooks.slack.com/services/T[a-zA-Z0-9_]{8}/B[a-zA-Z0-9_]{8}/[a-zA-Z0-9_]{24})")),
+    ),
+    SecretRegex(
+        finding_type="aws",
+        regex=re.compile(str(r"(AKIA[A-Z0-9]{16})")),
+    ),
+    SecretRegex(
+        finding_type="aws",
+        regex=re.compile(str(r"(ASIA[A-Z0-9]{16})")),
+    ),
+    SecretRegex(
+        finding_type="aws",
+        regex=re.compile(str(r"(ACCA[A-Z0-9]{16})")),
+    ),
+    SecretRegex(
+        finding_type="facebook",
+        regex=re.compile(str(r"(EAACEdEose0cBA[0-9A-Za-z]{16,255})")),
+    ),
+    SecretRegex(
+        finding_type="facebook",
+        regex=re.compile(str(r"([f|F][a|A][c|C][e|E][b|B][o|O][o|O][k|K](.){1,64}['|\"][0-9a-f]{32}['|\"])")),
+    ),
+    SecretRegex(
+        finding_type="Generic API Key",
+        regex=re.compile(str(r"([a|A][p|P][i|I][_]?[k|K][e|E][y|Y](.){1,64}['|\"][0-9a-zA-Z]{32,45}['|\"])")),
+    ),
+    SecretRegex(
+        finding_type="Generic Secret",
+        regex=re.compile(str(r"([s|S][e|E][c|C][r|R][e|E][t|T](.){1,64}['|\"][0-9a-zA-Z]{32,45}['|\"])")),
+    ),
+    SecretRegex(
+        finding_type="google",
+        regex=re.compile(str(r"([0-9]+-[0-9A-Za-z_]{32}\.apps\.googleusercontent\.com)")),
+    ),
+    SecretRegex(
+        finding_type="google",
+        regex=re.compile(str(r"(ya29\.[0-9A-Za-z\-_]{55})")),
+    ),
+    SecretRegex(
+        finding_type="mongo",
+        regex=re.compile(str(r"((mongodb|mongodb\+srv)" + URL_REGEX)),
+    ),
+    SecretRegex(
+        finding_type="postgres",
+        regex=re.compile(str(r"((postgres)" + URL_REGEX)),
+    ),
+    SecretRegex(
+        finding_type="urlauth",
+        regex=re.compile(str(r"((ftp|ftps|http|https)" + URL_REGEX)),
+    ),
```

Comment:
These three are collapsed into one rule when we run gitsecrets itself. We break them out, but it should be equivalent.